Many packages used in computer animation industry

guibou opened the pull request:
###### Motivation for this change

All of these packages are standard in rendering / animation / cinema industry:

- ptex : texture loading
- alembic : asset description
- partio : particules description
- openfx : post processing filters
- openexrid : post processing
- openvdb : voxel modelisation

We are using nix to build our product: http://guerillarender.com/ which depends on theses libs. I wanted to upstream this work.

These packages are created with our setup in mind, for example, I did not included the openvdb binaries or documentation. I'm motivated to take maintainership of these packages and improve them as needed.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

guibou committed:
Add *guibou* in maintainers

guibou committed:
ptex: init at 2.1.28

guibou committed:
alembic: init at 1.7.4

guibou committed:
partio: init at 1.1.0

guibou committed:
openexrid-unstable: init at 2017-09-17

guibou committed:
openfx: init at 1.4

guibou committed:
openvdb: init at 4.0.2

grahamc commented on pkgs/development/libraries/openexrid/default.nix:
```diff
@@ -0,0 +1,48 @@
+  version = "2017-09-17";
+
+  src = fetchurl {
+    url = "https://github.com/MercenariesEngineering/openexrid/archive/bec0081548a096f9bcdd1504970c96264b0fc050.zip";
```
These fetches may be better with fetchFromGitHub

guibou replied:
And actually it was written inside `CONTRIBUTING.md`. Sorry, I'm fixing that.

guibou commented:
I corrected the `fetchurl` to `fetchFromGitHub` and rebuilt everything. The travis CI was also complaining about a type in `license` versus `licenses` which is now fixed.

guibou commented:
The build failed for macOS integration. I don't have the resource to work this out (i.e: a computer with macOS). Should I change the architecture of the erroneous package or should we tag it as "broken" in some way?

rycee commented on pkgs/development/libraries/openexrid/default.nix:
```diff
@@ -0,0 +1,50 @@
+
+stdenv.mkDerivation rec
+{
+  name = "openexrid-${version}";
```
Per the [package naming guidelines](https://nixos.org/nixpkgs/manual/#sec-package-naming) I believe the name here should be `openexrid-unstable-${version}`.

guibou replied:
Corrected. Thank you.

edolstra commented:
Looks good to me, thanks!

vyp commented on pkgs/development/libraries/alembic/default.nix:
```diff
@@ -0,0 +1,50 @@
+    license = {
+      fullName = "Alembic License";
+      url = "https://github.com/alembic/alembic/blob/master/LICENSE.txt";
+    };
```
This is bsd3 license is it not?

vyp commented on pkgs/development/libraries/openfx/default.nix:
```diff
@@ -0,0 +1,41 @@
+    license = {
+      fullName = "OpenFX license";
+      url = "https://github.com/ofxa/openfx/blob/master/Support/LICENSE";
+    };
```
bsd3?

vyp commented on pkgs/development/libraries/partio/default.nix:
```diff
@@ -0,0 +1,55 @@
+    license = {
+      fullName = "Partio software";
+      url = "http://wdas.github.io/partio/doxygen/license.html";
+    };
```
bsd3?

vyp commented on pkgs/development/libraries/ptex/default.nix:
```diff
@@ -0,0 +1,48 @@
+    license = {
+      fullName = "Ptex License";
+      url = "http://ptex.us/documentation.html";
+    };
```
bsd3?

guibou commented:
As requested by @vyp, I changed the license to bsd3 when applicable.
